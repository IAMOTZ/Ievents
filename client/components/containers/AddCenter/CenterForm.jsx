import React from 'react';
import PropTypes from 'prop-types';
import ImageInput from '../../common/ImageInput';
import { LoadingIcon } from '../../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../../common/Alert';

const CenterForm = (props) => {
  const toUpdate = { ...props.toUpdate };
  return (
    <form className="mt-lg-5 mb-md-5">
      <LoadingIcon start={props.addingCenterStarted || props.updatingCenterStarted} size={3} />
      <div className="w-lg-50 mx-auto">
        <BigAlert message={props.addingCenterError || props.updatingCenterError} />
      </div>
      <div className="container">
        <div className="row">
          <div className="col-12 col-lg-6">
            <div className="form-group">
              <label htmlFor="name">Name</label>
              <input
                type="text"
                className="form-control"
                id="name"
                name="name"
                placeholder="The name of the center"
                defaultValue={toUpdate.name}
                onChange={props.getInput}
              />
              <small className="form-text text-muted">Between 2 and 30 characters</small>
              <SmallAlert message={props.inputErrors.nameError} />
            </div>
          </div>
          <div className="col-12 col-lg-6">
            <div className="form-group">
              <label htmlFor="location">Location</label>
              <input
                type="text"
                className="form-control"
                id="location"
                name="location"
                placeholder="The location of the center"
                defaultValue={toUpdate.location}
                onChange={props.getInput}
              />
              <small className="form-text text-muted">Less than 50 characters</small>
              <SmallAlert message={props.inputErrors.locationError} />
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-12 col-lg-6">
            <div className="form-group">
              <label htmlFor="details">Details</label>
              <textarea
                className="form-control"
                id="details"
                rows="7"
                name="details"
                placeholder="More details about the center"
                defaultValue={toUpdate.details}
                onChange={props.getInput}
              />
              <small className="form-text text-muted">Less than 300 characters</small>
              <SmallAlert message={props.inputErrors.detailsError} />
            </div>
          </div>
          <div className="col-12 col-lg-6">
            <div className="row">
              <div className="col-12">
                <div className="form-group">
                  <label htmlFor="capacity">Capacity</label>
                  <input
                    type="number"
                    pattern="\d*"
                    className="form-control"
                    id="capacity"
                    name="capacity"
                    placeholder="How many seats"
                    defaultValue={toUpdate.capacity}
                    onChange={props.getInput}
                  />
                  <SmallAlert message={props.inputErrors.capacityError} />
                </div>
              </div>
            </div>
            <div className="row">
              <div className="col-12">
                <div className="form-group">
                  <label htmlFor="price">Price</label>
                  <input
                    type="number"
                    className="form-control"
                    id="price"
                    name="price"
                    placeholder="Price"
                    defaultValue={toUpdate.price}
                    onChange={props.getInput}
                  />
                  <SmallAlert message={props.inputErrors.priceError} />
                </div>
              </div>
            </div>
            <div className="row">
              <div className="col-12">
                <div className="form-group">
                  <label htmlFor="image">Image</label>
                  <div className="text-center">
                    {/* <ImageInput
                      style={{ height: '100px' }}
                      id="image"
                      onDrop={props.handleImageDrop}
                      newImage={props.images ? props.images[0] : null}
                      previousImage={
                        toUpdate.images ? toUpdate.images[0] : null
                      }
                    /> */}
                  </div>
                </div>
              </div>
              <div className="ml-3 pt-3">
                <button
                  id="add-btn"
                  className="btn ie-blue-button"
                  disabled={props.addingCenterStarted || props.updatingCenterStarted}
                  onClick={props.add || props.update}
                >{toUpdate.name ? 'Update' : 'Add'}
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  );
};

CenterForm.defaultProps = {
  addingCenterStarted: undefined,
  addingCenterError: '',
  updatingCenterStarted: undefined,
  updatingCenterError: '',
  inputErrors: {},
  images: [],
  toUpdate: {},
  add: undefined,
  update: undefined,
};

/* eslint-disable react/forbid-prop-types */
CenterForm.propTypes = {
  addingCenterStarted: PropTypes.bool,
  addingCenterError: PropTypes.string,
  updatingCenterStarted: PropTypes.bool,
  updatingCenterError: PropTypes.string,
  getInput: PropTypes.func.isRequired,
  inputErrors: PropTypes.object,
  images: PropTypes.array,  
  handleImageDrop: PropTypes.func.isRequired,
  toUpdate: PropTypes.object,
  add: PropTypes.func,
  update: PropTypes.func,
};

export default CenterForm;
