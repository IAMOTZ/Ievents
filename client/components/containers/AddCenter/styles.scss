#add-center-container {
  .form-inputs {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
  }
  .text-inputs {
    display: flex;
    flex-flow: column wrap;
    min-width: 351px;
  }
}