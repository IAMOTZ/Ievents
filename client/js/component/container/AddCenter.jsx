import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
// Actions.
import { addCenter, clearStatus } from '../../actions/centerActions';
// Common Components.
import UserSideNav from '../common/SideNavigation.jsx';
import Header from '../common/Header.jsx';
import ImageInput from '../common/ImageInput.jsx';
import WarningAlert from '../common/WarningAlert.jsx';
import { UserTopNav } from '../common/TopNavigation.jsx';
import { LoadingIcon } from '../common/LoadingAnimation.jsx';

@connect(store => (
  {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    isAdmin: (store.user.user.role === 'admin'),
    isSuperAdmin: (store.user.user.role === 'superAdmin'),
    status: {
      error: store.centers.status.addingError.message,
      success: store.centers.status.added,
      adding: store.centers.status.adding,
    },
  }
))
export default class AddCenter extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      location: null,
      details: null,
      capacity: null,
      price: null,
      type: null,
      images: null,
    };
  }

  componentWillUnmount() {
    this.props.dispatch(clearStatus('ALL'));
  }

  /**
   * Update some state variables with the user inputs.
   * @param {Event} e The event object.
   */
  getInput = (e) => {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  /**
   * A callback to update the state after the user uploads an image.
   * @param {File} files The image file.
   */
  handleImageDrop = (files) => {
    this.setState({ images: files });
  }

  /**
   * Dispatches the action to add the center.
   */
  add = () => {
    const {
      name, location, details, capacity, price, images,
    } = this.state;
    const centerDetails = {
      name, location, details, capacity, price,
    };
    const fd = new FormData();
    const entries = Object.entries(centerDetails);
    entries.forEach((entry) => {
      const key = entry[0];
      const value = entry[1];
      if (value) {
        fd.append(key, value);
      }
    });
    if (images) {
      fd.append('image', images[0]);
    }
    this.props.dispatch(addCenter(fd, this.props.user.token));
    window.scrollTo(0, 0);
  }

  render() {
    let component;
    if (!this.props.authenticated) {
      component = (<Redirect to="/users/login" />);
    } else if (this.props.status.success) {
      component = (<Redirect to="/centers2" />);
    } else {
      component = (
        <div className="add-center-container">
          {/* Top navigation on small screen */}
          <UserTopNav
            name={this.props.user.name}
            title="Add a center"
            isAdmin={this.props.isAdmin}
            isSuperAdmin={this.props.isSuperAdmin}
            dispatch={this.props.dispatch}
          />
          <div className="container-fluid">
            <div className="row">
              {/*  Side navigation on large screen */}
              <UserSideNav
                name={this.props.user.name}
                isAdmin={this.props.isAdmin}
                isSuperAdmin={this.props.isSuperAdmin}
                dispatch={this.props.dispatch}
              />
              {/* Main content */}
              <div className="col-lg-10 offset-md-2" id="add-event-section">
                {/* Content Header(navigation) on large screen */}
                <Header text="Add a center" />
                {/* Input form */}
                <form className="mt-lg-5 mb-md-5">
                  <LoadingIcon start={this.props.status.adding} size={2} />
                  <div className="w-lg-50 mx-auto">
                    <WarningAlert message={this.props.status.error} />
                  </div>
                  <div className="container">
                    <div className="row">
                      <div className="col-12 col-lg-6">
                        <div className="form-group">
                          <label htmlFor="name">Name</label>
                          <input
                            type="email"
                            className="form-control"
                            id="name"
                            name="name"
                            placeholder="The name of the center"
                            onChange={this.getInput}
                          />
                          <small className="form-text text-muted">Less than 30 characters</small>
                        </div>
                      </div>
                      <div className="col-12 col-lg-6">
                        <div className="form-group">
                          <label htmlFor="location">Location</label>
                          <input
                            type="text"
                            className="form-control"
                            id="location"
                            name="location"
                            placeholder="The location of the center"
                            onChange={this.getInput}
                          />
                          <small className="form-text text-muted">Less than 30 characters</small>
                        </div>
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-12 col-lg-6">
                        <div className="form-group">
                          <label htmlFor="details">Details</label>
                          <textarea
                            className="form-control"
                            id="details"
                            rows="7"
                            name="details"
                            placeholder="More details about the center"
                            onChange={this.getInput}
                          />
                          <small className="form-text text-muted">Less than 300 characters</small>
                        </div>
                      </div>
                      <div className="col-12 col-lg-6">
                        <div className="row">
                          <div className="col-12">
                            <div className="form-group">
                              <label htmlFor="capacity">Capacity</label>
                              <input
                                type="number"
                                className="form-control"
                                id="capacity"
                                name="capacity"
                                placeholder="How many seats"
                                onChange={this.getInput}
                              />
                            </div>
                          </div>
                        </div>
                        <div className="row">
                          <div className="col-12">
                            <div className="form-group">
                              <label htmlFor="price">Price</label>
                              <input
                                type="number"
                                className="form-control"
                                id="price"
                                name="price"
                                placeholder="Price"
                                onChange={this.getInput}
                              />
                            </div>
                          </div>
                        </div>
                        <div className="row">
                          <div className="col-12">
                            <div className="form-group">
                              <label htmlFor="image">Image</label>
                              <div className="text-center">
                                <ImageInput
                                  style={{ height: '100px' }}
                                  id="image"
                                  onDrop={this.handleImageDrop}
                                  newImage={this.state.images ? this.state.images[0] : null}
                                />
                              </div>
                            </div>
                          </div>
                          <div className="ml-3 pt-3">
                            <button
                              id="add-btn"
                              className="btn btn-outline-dark"
                              disabled={this.props.status.adding}
                              onClick={this.add}
                            >Add
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <footer className="d-block d-sm-none mt-5">
            <div className="container text-white text-center py-5">
              <h1>Ievents</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </footer>
        </div>
      );
    }
    return component;
  }
}
