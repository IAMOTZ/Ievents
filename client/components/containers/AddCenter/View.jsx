import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import { AuthTopNavigation } from '../../common/TopNavigation';
import CenterForm from './CenterForm';

const View = props => (
  <div className="add-center-container">
    <AuthTopNavigation
      name={props.userName}
      title="Add a center"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-md-2" id="add-event-section">
          <Header text="Add a center" />
          <CenterForm
            addingCenterStarted={props.addingCenterStarted}
            addingCenterError={props.addingCenterError}
            getInput={props.getInput}
            inputErrors={props.inputErrors}
            handleImageDrop={props.handleImageDrop}
            images={props.images}
            add={props.add}
          />
        </div>
      </div>
    </div>
    <footer className="d-block d-sm-none mt-5">
      <div className="container text-white text-center py-5">
        <h1>Ievents</h1>
        <p>Copyright &copy; 2017</p>
      </div>
    </footer>
  </div>
);

View.defaultProps = {
  addingCenterError: '',
  inputErrors: {},
  images: [],
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  addingCenterStarted: PropTypes.bool.isRequired,
  addingCenterError: PropTypes.string,
  inputErrors: PropTypes.object,
  images: PropTypes.array,
  add: PropTypes.func.isRequired,
  getInput: PropTypes.func.isRequired,
  handleImageDrop: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default View;
