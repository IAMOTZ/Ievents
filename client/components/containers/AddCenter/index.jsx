import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { validateAddCenterInputs } from '../../../helpers/inputValidators';
import { addCenter } from '../../../actions/centerActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import View from './View';


@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    addingCenterStarted: store.addCenterReducer.addingCenterStarted,
    addingCenterResolved: store.addCenterReducer.addingCenterResolved,
    addingCenterError: store.addCenterReducer.addingCenterError,
  };
})
class AddCenter extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      location: null,
      details: null,
      capacity: null,
      price: null,
      images: null,
      inputErrors: {
        nameError: null,
        locationError: null,
        detailsError: null,
        capacityError: null,
        priceError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_CENTER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      nameError: null,
      locationError: null,
      detailsError: null,
      capacityError: null,
      priceError: null,
    };
    this.setState(state);
  }

  /**
   * A callback to update the state after the user uploads an image.
   * @param {File} files The image file.
   */
  handleImageDrop = (files) => {
    this.setState({ images: files });
  }

  /**
   * Dispatches the action to add the center.
   * @param {Event} event The event object.
   */
  add = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_CENTER));
    const {
      name, location, details, capacity, price, images,
    } = this.state;
    const centerDetails = {
      name, location, details, capacity, price,
    };
    const inputErrors = validateAddCenterInputs(centerDetails);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      const fd = new FormData();
      const entries = Object.entries(centerDetails);
      entries.forEach((entry) => {
        const key = entry[0];
        const value = entry[1];
        if (value) {
          fd.append(key, value);
        }
      });
      if (images) {
        fd.append('image', images[0]);
      }
      this.props.dispatch(addCenter(fd, this.props.userToken));
    }
    window.scrollTo(0, 0);
  }

  render() {
    if (this.props.addingCenterResolved) {
      return (<Redirect to="/centers2" />);
    }
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        addingCenterStarted={this.props.addingCenterStarted}
        addingCenterError={this.props.addingCenterError}
        getInput={this.getInput}
        inputErrors={this.state.inputErrors}
        handleImageDrop={this.handleImageDrop}
        images={this.state.images}
        add={this.add}
      />
    );
  }
}

AddCenter.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  addingCenterStarted: PropTypes.bool,
  addingCenterResolved: PropTypes.bool,
  addingCenterError: PropTypes.string,
  dispatch: PropTypes.func,
};

AddCenter.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  addingCenterStarted: false,
  addingCenterResolved: false,
  addingCenterError: '',
  dispatch: () => {},
};

export default AddCenter;
