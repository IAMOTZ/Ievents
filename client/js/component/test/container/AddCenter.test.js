/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import AddCenter from '../../container/AddCenter.jsx';

const props = {
  user: { name: 'test-user' },
  authenticated: true,
  isAdmin: true,
  isSuperAdmin: false,
  status: {
    adding: false,
    success: false,
    error: false,
  },
  dispatch: () => { },
};

const alterProps = newProps => ({
  user: newProps.user || props.user,
  authenticated: newProps.authenticated === undefined ? true : newProps.authenticated,
  isAdmin: newProps.isAdmin === undefined ? true : newProps.isAdmin,
  isSuperAdmin: newProps.isSuperAdmin || false,
  status: {
    adding: newProps.adding || false,
    success: newProps.success || false,
    error: newProps.error || false,
  },
  dispatch: () => { },
});

describe('<AddCenter />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AddCenter {...props} />);
  })
  describe('Rendering:', () => {
    it('should render correctly on page load', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to login page if user is not authenticated', () => {
      wrapper.setProps(alterProps({ authenticated: false }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to centers page if center is successfully added', () => {
      wrapper.setProps(alterProps({ success: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correcty when adding', () => {
      wrapper.setProps(alterProps({ adding: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when there is an error', () => {
      wrapper.setProps(alterProps({ error: 'there was an error' }));
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AddCenter {...props} />);
    });
    it('should update the state when the user keys in the center name', () => {
      const name = 'test-center';
      wrapper
        .find('#name')
        .simulate('change', { target: { name: 'name', value: name } });
      expect(wrapper.state('name')).toEqual(name);
    });
    it('should update the state when the user keys in the center location', () => {
      const location = 'test-location';
      wrapper
        .find('#location')
        .simulate('change', { target: { name: 'location', value: location } });
      expect(wrapper.state('location')).toEqual(location);
    });
    it('should update the state when the user keys in the center details', () => {
      const details = 'test-details';
      wrapper
        .find('#details')
        .simulate('change', { target: { name: 'details', value: details } });
      expect(wrapper.state('details')).toEqual(details);
    });
    it('should update the state when the user keys in the center capacity', () => {
      const capacity = 100;
      wrapper
        .find('#capacity')
        .simulate('change', { target: { name: 'capacity', value: capacity } });
      expect(wrapper.state('capacity')).toEqual(capacity);
    });
    it('should update the state when the user keys in the center price', () => {
      const price = 200;
      wrapper
        .find('#price')
        .simulate('change', { target: { name: 'price', value: price } });
      expect(wrapper.state('price')).toEqual(price);
    });
  });
});
