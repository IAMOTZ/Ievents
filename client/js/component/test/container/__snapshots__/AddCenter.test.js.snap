// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCenter /> Rendering: should redirect to centers page if center is successfully added 1`] = `
<Redirect
  push={false}
  to="/centers2"
/>
`;

exports[`<AddCenter /> Rendering: should redirect to login page if user is not authenticated 1`] = `
<Redirect
  push={false}
  to="/users/login"
/>
`;

exports[`<AddCenter /> Rendering: should render correctly on page load 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Add a center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Add a center"
        />
        <form
          className="mt-lg-5 mb-md-5"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <div
            className="w-lg-50 mx-auto"
          >
            <WarningAlert
              message={false}
            />
          </div>
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="name"
                  >
                    Name
                  </label>
                  <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="The name of the center"
                    type="email"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="location"
                  >
                    Location
                  </label>
                  <input
                    className="form-control"
                    id="location"
                    name="location"
                    onChange={[Function]}
                    placeholder="The location of the center"
                    type="text"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="details"
                  >
                    Details
                  </label>
                  <textarea
                    className="form-control"
                    id="details"
                    name="details"
                    onChange={[Function]}
                    placeholder="More details about the center"
                    rows="7"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 300 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="capacity"
                      >
                        Capacity
                      </label>
                      <input
                        className="form-control"
                        id="capacity"
                        name="capacity"
                        onChange={[Function]}
                        placeholder="How many seats"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="price"
                      >
                        Price
                      </label>
                      <input
                        className="form-control"
                        id="price"
                        name="price"
                        onChange={[Function]}
                        placeholder="Price"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="image"
                      >
                        Image
                      </label>
                      <div
                        className="text-center"
                      >
                        <ImageInput
                          id="image"
                          newImage={null}
                          onDrop={[Function]}
                          style={
                            Object {
                              "height": "100px",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="ml-3 pt-3"
                  >
                    <button
                      className="btn btn-outline-dark"
                      disabled={false}
                      id="add-btn"
                      onClick={[Function]}
                    >
                      Add
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright Â© 2017
      </p>
    </div>
  </footer>
</div>
`;

exports[`<AddCenter /> Rendering: should render correctly when there is an error 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Add a center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Add a center"
        />
        <form
          className="mt-lg-5 mb-md-5"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <div
            className="w-lg-50 mx-auto"
          >
            <WarningAlert
              message="there was an error"
            />
          </div>
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="name"
                  >
                    Name
                  </label>
                  <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="The name of the center"
                    type="email"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="location"
                  >
                    Location
                  </label>
                  <input
                    className="form-control"
                    id="location"
                    name="location"
                    onChange={[Function]}
                    placeholder="The location of the center"
                    type="text"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="details"
                  >
                    Details
                  </label>
                  <textarea
                    className="form-control"
                    id="details"
                    name="details"
                    onChange={[Function]}
                    placeholder="More details about the center"
                    rows="7"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 300 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="capacity"
                      >
                        Capacity
                      </label>
                      <input
                        className="form-control"
                        id="capacity"
                        name="capacity"
                        onChange={[Function]}
                        placeholder="How many seats"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="price"
                      >
                        Price
                      </label>
                      <input
                        className="form-control"
                        id="price"
                        name="price"
                        onChange={[Function]}
                        placeholder="Price"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="image"
                      >
                        Image
                      </label>
                      <div
                        className="text-center"
                      >
                        <ImageInput
                          id="image"
                          newImage={null}
                          onDrop={[Function]}
                          style={
                            Object {
                              "height": "100px",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="ml-3 pt-3"
                  >
                    <button
                      className="btn btn-outline-dark"
                      disabled={false}
                      id="add-btn"
                      onClick={[Function]}
                    >
                      Add
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright Â© 2017
      </p>
    </div>
  </footer>
</div>
`;

exports[`<AddCenter /> Rendering: should render correcty when adding 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Add a center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Add a center"
        />
        <form
          className="mt-lg-5 mb-md-5"
        >
          <LoadingIcon
            size={2}
            start={true}
          />
          <div
            className="w-lg-50 mx-auto"
          >
            <WarningAlert
              message={false}
            />
          </div>
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="name"
                  >
                    Name
                  </label>
                  <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="The name of the center"
                    type="email"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="location"
                  >
                    Location
                  </label>
                  <input
                    className="form-control"
                    id="location"
                    name="location"
                    onChange={[Function]}
                    placeholder="The location of the center"
                    type="text"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 30 characters
                  </small>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="details"
                  >
                    Details
                  </label>
                  <textarea
                    className="form-control"
                    id="details"
                    name="details"
                    onChange={[Function]}
                    placeholder="More details about the center"
                    rows="7"
                  />
                  <small
                    className="form-text text-muted"
                  >
                    Less than 300 characters
                  </small>
                </div>
              </div>
              <div
                className="col-12 col-lg-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="capacity"
                      >
                        Capacity
                      </label>
                      <input
                        className="form-control"
                        id="capacity"
                        name="capacity"
                        onChange={[Function]}
                        placeholder="How many seats"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="price"
                      >
                        Price
                      </label>
                      <input
                        className="form-control"
                        id="price"
                        name="price"
                        onChange={[Function]}
                        placeholder="Price"
                        type="number"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-12"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="image"
                      >
                        Image
                      </label>
                      <div
                        className="text-center"
                      >
                        <ImageInput
                          id="image"
                          newImage={null}
                          onDrop={[Function]}
                          style={
                            Object {
                              "height": "100px",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="ml-3 pt-3"
                  >
                    <button
                      className="btn btn-outline-dark"
                      disabled={true}
                      id="add-btn"
                      onClick={[Function]}
                    >
                      Add
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright Â© 2017
      </p>
    </div>
  </footer>
</div>
`;
